x-pack/plugins/observability_solution/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -119,7 +118,12 @@ export const createCustomThresholdExecutor = ({
     const groupByIsSame = isEqual(state.groupBy, params.groupBy);
     const previousMissingGroups =
       alertOnGroupDisappear && queryIsSame && groupByIsSame && state.missingGroups
-        ? state.missingGroups
+        ? state.missingGroups.filter((missingGroup) =>
+            // We use isTrackedAlert to remove missing groups that are untracked by the user
+            typeof missingGroup === 'string'
```

Comment:
@maryam-saeidi I think we can remove this check since `missingGroup` is always `MissingGroupsRecord` now.